plugins/Rockets/jsonPropertyMap.h

```diff
@@ -552,25 +753,128 @@ inline bool from_json(brayns::PropertyMap& obj, const std::string& json)
             return false;
         switch (obj.getPropertyType(propName))
         {
-        case PropertyMap::Property::Type::Float:
-            SET_PROPERTY_NUMBER(Double)
+        case PropertyMap::Property::Type::Double:
+        {
+            double val;
+            if (!get_property(m.value, val))
+                return false;
+            obj.updateProperty(propName, val);
+            break;
+        }
         case PropertyMap::Property::Type::Int:
-            SET_PROPERTY(Int)
+            if (!parseSetProperty<int32_t>(m.value, obj, propName))
+                return false;
+            break;
         case PropertyMap::Property::Type::String:
-            SET_PROPERTY(String)
+            if (!parseSetProperty<std::string>(m.value, obj, propName))
+                return false;
+            break;
         case PropertyMap::Property::Type::Bool:
-            SET_PROPERTY(Bool)
-        case PropertyMap::Property::Type::Vec2f:
-            SET_ARRAY(double, Double, 2)
+            if (!parseSetProperty<bool>(m.value, obj, propName))
+                return false;
+            break;
+        case PropertyMap::Property::Type::Vec2d:
+            if (!parseSetArray<double, 2>(m.value, obj, propName))
+                return false;
+            break;
         case PropertyMap::Property::Type::Vec2i:
-            SET_ARRAY(int32_t, Int, 2)
-        case PropertyMap::Property::Type::Vec3f:
-            SET_ARRAY(double, Double, 3)
+            if (!parseSetArray<int32_t, 2>(m.value, obj, propName))
+                return false;
+            break;
+        case PropertyMap::Property::Type::Vec3d:
+            if (!parseSetArray<double, 3>(m.value, obj, propName))
+                return false;
+            break;
         case PropertyMap::Property::Type::Vec3i:
-            SET_ARRAY(int32_t, Int, 3)
-        case PropertyMap::Property::Type::Vec4f:
-            SET_ARRAY(double, Double, 4)
+            if (!parseSetArray<int32_t, 3>(m.value, obj, propName))
+                return false;
+            break;
+        case PropertyMap::Property::Type::Vec4d:
+            if (!parseSetArray<double, 4>(m.value, obj, propName))
+                return false;
+            break;
         }
     }
     return true;
 }
+
+template <>
+inline std::string to_json(const brayns::ModelProperties& props)
```

Comment:
a comment why this code is here (whatever this does), should be there.

Reply:
It should not be here. Done.